### PR TITLE
Remove `doHandshake` test-only settings from TcpTransport

### PR DESCRIPTION
**Pull request by s1monw**
In #22094 we introduce a test-only setting to simulate transport
impls that don't support handshakes. This commit implements the same logic
without a setting.

### CODE REVIEW

**Comment by s1monw** [comment]
@elasticmachine test this please

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by s1monw** [comment]
@elasticmachine test this please

File: core/src/main/java/org/elasticsearch/transport/TcpTransport.java [hunk 1]
```diff
@@ -463,21 +458,13 @@ public void connectToNode(DiscoveryNode node, ConnectionProfile connectionProfil
...
-                        if (version != null) {
-                            // this is a BWC layer, if we talk to a pre 5.2 node then the handshake is not supported
-                            // this will go away in master once it's all ported to 5.2 but for now we keep this to make
-                            // the backport straight forward
-                            nodeChannels = new NodeChannels(nodeChannels, version);
-                        }
-                    }
+                    Channel channel = nodeChannels.channel(TransportRequestOptions.Type.PING);
+                    final TimeValue connectTimeout = connectionProfile.getConnectTimeout() == null ?
+                        defaultConnectionProfile.getConnectTimeout():
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/transport/TcpTransport.java]: Nit: `():` -> `() :` (whitespace)

**Review by jasontedor — APPROVED** [review]
Thanks for doing this @s1monw, it looks great.

**Pull request merged**